File: js/image.js
CanvasImage = function(imgElement) {
	this.isCircle = this.imgElement == "Circle";
	this.isRect = this.imgElement == "Rect";
	this.imgElement = imgElement;

	this.x = 0, this.y = 0;
	this.width = 0, this.height = 0;
	this.angle = 0;
	this.color = "black";
}

CanvasImage.prototype.draw = function() {
	if (this.isCircle) {
		ctx.beginPath();
		ctx.arc(this.x, this.y, this.height, 0, 2 * Math.PI, false);
		ctx.fillStyle = this.color;
		ctx.fill();
	} else {
		ctx.save();
		ctx.translate(this.x, this.y);
		ctx.rotate(toRadians * this.angle);

		ctx.drawImage(this.imgElement, -this.width / 2, -this.height / 2, this.width, this.height);
		ctx.restore();
	}
}

File: dart/image.dart
import 'dart:html';
import 'dart:math';

class Image {
	static var images = [];

	ImageElement img;
	var ctx;
	var x, y;
	var width, height;
	var angle;
	var scale = 0.3;

	Image(img, context) {
		ctx = context;

		this.img = img;
	}

	void draw() {
		var toRadians = PI / 180; 
		ctx.save();
		ctx.translate(x, y);
		ctx.rotate(toRadians * angle);

		ctx.drawImageScaled(img, -width / 2, -height / 2, width, height);
		ctx.restore();
	}
}
